tests/test_pipeline.py

```diff
@@ -0,0 +1,96 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import OrderedDict
+
+import numpy as np
+
+import pytest
+from deepsparse.image_classification import (
+    IMAGENET_RGB_MEANS,
+    IMAGENET_RGB_STDS,
+    ImageClassificationInput,
+    ImageClassificationOutput,
+)
+from deepsparse.pipeline import FunctionPipeline
+from sparsezoo import Zoo
+from sparsezoo.utils import load_numpy_list
+
+
+from PIL import Image  # isort:skip
+from torchvision import transforms  # isort:skip
+
+
+@pytest.mark.parametrize(
+    "zoo_stub,image_size",
+    [
+        (
+            "zoo:cv/classification/resnet_v1-50/pytorch/sparseml/imagenet/"
+            "pruned85_quant-none-vnni",
+            224,
+        )
+    ],
+)
+def test_function_pipeline_as_image_classifier(zoo_stub, image_size):
+    non_rand_resize_scale = 256.0 / 224.0  # standard used
+    standard_imagenet_transforms = transforms.Compose(
+        [
+            transforms.ToTensor(),
+            transforms.Resize(round(non_rand_resize_scale * image_size)),
+            transforms.CenterCrop(image_size),
+            transforms.Normalize(mean=IMAGENET_RGB_MEANS, std=IMAGENET_RGB_STDS),
+        ]
+    )
+
+    def preprocess(input: ImageClassificationInput):
+        assert len(input.images) == 1
+        assert isinstance(input, ImageClassificationInput)
+        assert isinstance(input.images, list)
+        assert all(isinstance(i, np.ndarray) for i in input.images)
+        return [
+            standard_imagenet_transforms(Image.fromarray(img)).unsqueeze(0).numpy()
+            for img in input.images
+        ]
+
+    def postprocess(outputs):
+        # assert len(outputs) == 1
```

Comment:
Oh for some reason there are 2 outputs even though there's only 1 input image. Did I do something wrong?

Reply:
the 2 outputs are both numpy arrays, but you're saying it could also have a dictionary output as well?

Reply:
ah shoot, yeah. our IC models do have two outputs. the logits before/after softmax

Reply:
preferring the second for postprocess is safe